docs/design/README.md

```diff
@@ -0,0 +1,552 @@
+The command creates a directory structure like the one shown above.
+The new project will include a minimal variant with a simple custom package and settings.
+The command will take a few arguments, such as `--project name` and `--variant-name`.
+
+Similarly, a command will be available to add a new variant to an existing project.
+For example, `twoliter new-variant` which will add a variant.
+
+```
+❯ twoliter new project --name ootb-variants --variant-name example-dev
+❯ cd ootb-variants
+❯ twoliter build --variant hello-dev
+  ...
+❯ twoliter new-variant my-new-variant
+```
+
+### Twoliter Update
+
+`twoliter update` is another inspiration from `cargo`.
+This command makes it easy for maintainers to update the packages (kits actually) they are using to the latest versions.
+
+For simplicity, no two variants in a project can use differing versions of a single kit.
+Twoliter will have the ability to create a kit dependency tree with the following procedure.
+First Twoliter will read the kit dependencies from each variant and add any external kits to its graph.
+Next Twoliter will read the kit definitions in the `kits` directory, and add all of their external dependencies to its graph.
+
+Once it has this list, Twoliter will begin a traversal.
+For each kit it will pull the `metdata` container for each external kit and store the kit.json files locally.
+Each time it finds a new external kit (i.e. an external kit that is depended on by an external kit), it will add this new dependency to the graph.
+(TODO: are cycles OK? They might be.)
+
+At this point Twoliter will determine the maximum version of each kit that is possible without introducing multiple versions of the same kit.
+TODO: this is probably hard, but there is prior art to look at.
+
+## Twoliter Build
+
+Here is a synopsis of how the build of a Bottlerocket out-of-tree proceeds.
+
+For this walkthrough, we will use the directory structure shown above.
+This means we will have the following items and dependencies:
+* Variant: `example-dev` which depends on the `hello-dev-kit` by placing it in the `Cargo.toml` `build-dependencies` section.
+* Kit: The `hello-dev-kit` depends on the `hello-agent` package by placing it in the `Cargo.toml` `build-dependencies` section.
+* Package: The `hello-agent` package.
+
+The user kicks things off with:
+
+```sh
+twoliter build --variant example-dev --arch aarch64
+```
+
+First Twoliter learns the external kit dependency graph using the traversal described for `twoliter update`.
+It pulls all kit container images that will be needed for the build.
+
+Twoliter starts the twoliter container environment and mounts the project directory.
+The container has a copy of Bottlerocket's `Makefile.toml` build tools in it.
+Twoliter invokes the `cargo make build` task.
+
+The `cargo make build` task is an alias of `cargo make variant` and its entrypoint is the variant's `Cargo.toml` file.
```

Comment:
Will `twoliter` allow users to use the exact same variables used to configure a build with the existing `Makefile.toml`?